lib/app/hooks/use_watch_once.dart

```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: ice License 1.0
+
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+/// A hook that listens to a Riverpod provider and only takes the first emitted value.
+///
+/// If [notNull] is true, the hook will ignore null values and wait for the first non-null value.
+T? useWatchOnce<T>(WidgetRef ref, ProviderListenable<T> provider, {bool notNull = true}) {
+  final value = useState<T?>(null);
+  final hasValue = useRef(false);
+
+  useEffect(
+    () {
+      if (!hasValue.value) {
+        final subscription = ref.listenManual<T>(provider, fireImmediately: true, (_, next) {
+          if (!hasValue.value && (!notNull || next != null)) {
```

Comment:
no need for !hasValue.value here cause checked in upper if

Reply:
yes, removed